Sources/PackageGraph/ResolvedPackage.swift

```diff
@@ -17,19 +17,31 @@ public final class ResolvedPackage: ObjectIdentifierProtocol {
     /// The underlying package reference.
     public let underlyingPackage: Package
 
+    // The identity of the package.
+    public var identity: PackageIdentity {
+        return self.underlyingPackage.identity
+    }
+
     /// The manifest describing the package.
     public var manifest: Manifest {
-        return underlyingPackage.manifest
+        return self.underlyingPackage.manifest
     }
 
     /// The name of the package.
+    @available(*, deprecated, message: "use identity (or manifestName, but only if you must) instead")
     public var name: String {
-        return underlyingPackage.name
+        return self.underlyingPackage.name
+    }
+
+    /// The name of the package as entered in the manifest.
+    /// This should rarely be used beyond presentation purposes
+    public var manifestName: String {
```

Comment:
Is this something we'll be able to move toward getting rid of once we have a strong notion of identity?

Reply:
yes! https://github.com/apple/swift-package-manager/pull/3323#discussion_r589835348